Small improvements to the Matter documentation (#28010)
home-assistant/home-assistant.github.io: source/_integrations/matter.markdown

For communicating with Matter devices, the Home Assistant integration runs its own "Matter controller" in a separate process which will be launched as an add-on. This add-on runs the controller software and connects your Matter network (called Fabric in technical terms) and Home Assistant. The Home Assistant Matter integration connects to this server via a WebSocket connection.

### Supported installation types

It is recommended to run the Matter add-on on Home Assistant OS. This is currently the best-supported option. 

> If you run Home Assistant in a container, you can run a docker image of the [Matter server](https://github.com/home-assistant-libs/python-matter-server). The requirements and instructions for your host setup are described on that github page.

Reviewer: ```suggestion
If you run Home Assistant in a container, you can run a Docker image of the [Matter server](https://github.com/home-assistant-libs/python-matter-server). The requirements and instructions for your host setup are described on that GitHub page.
```

For communicating with Matter devices, the Home Assistant integration runs its own "Matter controller" in a separate process which will be launched as an add-on. This add-on runs the controller software and connects your Matter network (called Fabric in technical terms) and Home Assistant. The Home Assistant Matter integration connects to this server via a WebSocket connection.

> The recommended configuration for the Matter integration is by running the officially provided Home Assistant Matter add-on on Home Assistant OS, due to various requirements to the host OS and configuration. For Homeassistant Container based installs, we also provide a docker image which you can run, see for instructions the repository of the [Matter server](https://github.com/home-assistant-libs/python-matter-server) including all requirements and instructions for your host setup. It is not possible to run Matter on Home Assistant Core only based installs.

Reviewer: ```suggestion
### Supported installation types

It is recommended to run the Matter add-on on Home Assistant OS. This is currently the best-supported option. 

If you run Home Assistant in a container, you can run a docker image of the [Matter server](https://github.com/home-assistant-libs/python-matter-server). The requirements and instructions for your host setup are described on that github page.

Running Matter on a Home Assistant Core installation is not supported.
```

[Eve Door & Window on Amazon](https://amzn.to/3RIU6ml)
[Eve Motion on Amazon](https://amzn.to/3jDujiP)

### Nanoleaf (Essentials) Matter bulbs and Lightstrips

- Although the products work great once commissioned, multiple users have reported that commissioning them can be a bit difficult and requires some patience and multiple resets or optimizations to your home network.
- Check the [Nanoleaf Matter infopage](https://nanoleaf.me/en-EU/integration/matter/) for all supported products and instructions.

### Philips Hue (V2) Bridge

> The Philips Hue V2 bridge supports Matter since a recent update. You can enable Matter support from the Hue app after which you can commission it to Home Assistant and other fabrics.

Reviewer: ```suggestion
The Philips Hue V2 bridge supports Matter. You can enable Matter support in the Hue app. You can then commission it to Home Assistant and other fabrics.
```

- Binding the Hue bridge to Home Assistant does not make sense because you will lose functionality over the default Hue integration in Home Assistant, such as button press events and (dynamic) scenes.
- You will need a Hue/Signify (cloud) account and the app before you can use Matter.
- Device events for example for dimmer remotes are not supported.
- Only basic control of lights is supported, no scenes, events, effects etc.

> ### Switchbot Hub 2

Reviewer: ```suggestion
### SwitchBot Hub 2
```

> Switchbot has released a (beta) firmware update to enable Matter support on their Hub 2, effectively bridging (some of the) devices on the Hub 2, such as curtain motors into Matter. The Switchbot Hub 2 is a Matter bridge device. 

Reviewer: ```suggestion
SwitchBot has released a (beta) firmware update to enable Matter support on their Hub 2.  The SwitchBot Hub 2 is a Matter bridge device. It is bridging some of the devices, such as curtain motors, into Matter.
```

> - You need to enable the Matter bridge support in the Switchbot app, copy the code and use that to commission the Hub to HA (or use a second device to scan the QR code).

Reviewer: ```suggestion
- To use Matter, in the SwitchBot app, enable Matter bridge support. Then, copy the code and use that to commission the Hub to Home Assistant. Another option is to use a second device to scan the QR code.
```

> - Device support is limited, you bridge specific devices to Matter by adding them as "Secondary device" in the app. Note that not all switchbot devices can be bridged.

Reviewer: ```suggestion
- Device support is limited. You bridge specific devices to Matter by adding them as **Secondary device** in the app. Note that not all SwitchBot devices can be bridged.
```

> - Enabling Matter support does not convert the actual swiotchbot devices into matter devices, those still need to be within the (bluetooth) range of the Hub.

Reviewer: ```suggestion
- Enabling Matter support does not convert the actual SwitchBot devices into matter devices. Those still need to be within the Bluetooth range of the hub.
```

> - Bridged Switchbot devices show up with a rather weird/technical name in Home Assistant. This is a known issue we've reported to Switchbot.

Reviewer: ```suggestion
- Bridged SwitchBot devices appear with a rather technical name in Home Assistant. This is a known issue.
```

### Tasmota

Tasmota supports Matter over IP on all ESP32 based devices (in experimental phase). Follow the [instructions](https://tasmota.github.io/docs/Matter/).

### TP-Link Tapo P125M (power plug)

> - TP-Link Tapo P125MLook for the M addition in the model name, a device without the M (regular P125) is not Matter compliant.

Reviewer: ```suggestion
- Look for the _M_ addition in the model name. A device without the M (regular P125) is not Matter compliant.
```

### General recommendations

> - Using Thread-based Matter devices in Home Assistant requires Home Assistant OS (version 10 and above) because of kernel patches to solve routing issues. Not using HAOS (and thus the official Matter add-on) is at your own risk but we do provide tips and tricks how to run the Matter Server as docker container yourself, including all requirements to the host and networking on the [matter server repository](https://github.com/home-assistant-libs/python-matter-server/blob/main/README.md)

Reviewer: ```suggestion
- Using Thread-based Matter devices in Home Assistant requires Home Assistant OS version 10 and above. Not using Home Assistant OS is at your own risk. We do provide some [documentation](https://github.com/home-assistant-libs/python-matter-server/blob/main/README.md) on how to run the Matter Server as a docker container. The documentation includes a description of the host and networking requirements. 
```

Author: well, a container install is supported since this month but with all precautions and warnings applied

Reviewer: Right. Let me rephrase. I hope it is more accurate now.

- To use Thread devices you will need a Thread Network with at least one Thread Border Router in your network nearby the Thread device(s). Apple users need for example the Apple TV 4K or the HomePod Mini, while Google users need a Nest Hub V2. Use the Thread integration in Home Assistant to diagnose your Thread network(s).

- Start simple and work from there, keep your network simple and add for example an ESP32 test device. Once that works, move on to the next step or more devices.

- Realize that you are an early adopter, both on the hardware side and on the software (controller) side so you may run into compatibility issues or features that are still missing. Report any issues you may find and help out others if you find a workaround or tested a device.

> - Make sure IPv6 (multicast) traffic travels freely from your network to the Home Assistant host. There is no requirement to have an IPv6-enabled internet connection or DHCPv6 server. However, IPv6 support has to be enabled (it's enabled by default on Home Assistant OS). 

Reviewer: ```suggestion
- Make sure IPv6 (multicast) traffic travels freely from your network to the Home Assistant host. There is no requirement to have an IPv6-enabled internet connection or DHCPv6 server. However, IPv6 support has to be enabled on Home Assistant. Go to **{% my network title="Settings > System > Network" %}**, and make sure **IPv6** is set to **Automatic** or **static**, depending on your network setup. If you're unsure, use **Automatic**.
```

> - See more advanced recommendations to host and network configuration in the [readme of the matter server repository](https://github.com/home-assistant-libs/python-matter-server/blob/main/README.md).

Reviewer: ```suggestion
- For more detailed information on network configuration, refer to the [README of the Matter server repository](https://github.com/home-assistant-libs/python-matter-server/blob/main/README.md).
```

### General recommendations

> - Using Thread-based Matter devices in Home Assistant requires Home Assistant OS version 10 and above. Not using Home Assistant OS is at your own risk. We do provide some [documentation](https://github.com/home-assistant-libs/python-matter-server/blob/main/README.md) on how to run the Matter Server as a docker container. The documentation includes a description of the host and networking requirements. 

Reviewer: ```suggestion
- Using Thread-based Matter devices in Home Assistant requires Home Assistant OS version 10 and above. Not using Home Assistant OS is at your own risk. We do provide some [documentation](https://github.com/home-assistant-libs/python-matter-server/blob/main/README.md) on how to run the Matter Server as a Docker container. The documentation includes a description of the host and networking requirements. 
```

The Matter protocol relies on (local) IPv6 and mDNS (multicast traffic) which should be able to travel freely in your network. Matter devices that use Wi-Fi (including Thread Border routers) must be on the same LAN/VLAN as Home Assistant. Matter devices that only use Thread must be joined to Thread networks for which there is at least one border router connected to the Home Assistant LAN.

> Investigate your network topology if you experience any issues with discovering devices (like the initial commission keeps failing) or if devices become unavailable randomly. For instance, a setting on your router or Wi-Fi access point to "optimize" multicast traffic can harm the (discovery) traffic from Matter devices. Keep this in mind when you experience issues trying to add or control Matter devices. Protocols like Matter are designed for regular residential network setups and do not play nicely with enterprise networking solutions like VLANs, Multicast filtering, and (malfunctioning) IGMP snooping. Try to keep your network as simple and flat as possible to avoid issues.

Reviewer: ```suggestion
If you experience any issues with discovering devices (for example, if the initial commission keeps failing or if devices become unavailable randomly), investigate your network topology. For instance, a setting on your router or Wi-Fi access point to "optimize" multicast traffic can harm the (discovery) traffic from Matter devices. Keep this in mind when you experience issues trying to add or control Matter devices. Protocols like Matter are designed for regular residential network setups and may not integrate well with enterprise networking solutions like VLANs, Multicast filtering, and (malfunctioning) IGMP snooping. To avoid issues, try to keep your network topology as simple and flat as possible.
```
